### PR TITLE
OCM-7347 | fix: allow to reset tuning configs

### PR DESCRIPTION
**Pull request by gdbranco**
[OCM-7347](https://issues.redhat.com//browse/OCM-7347)

**Change type**
- [ ] New feature
- [x] Bug fix
- [ ] Build
- [ ] CI
- [ ] Documentation
- [ ] Performance
- [ ] Refactor
- [ ] Style
- [ ] Unit tests
- [x] Subsystem tests

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: provider/machinepool/hcp/machine_pool_resource.go [hunk 1]
```diff
@@ -1186,7 +1190,9 @@ func populateState(ctx context.Context, object *cmv1.NodePool, state *HcpMachine
 		}
 		state.TuningConfigs = tuningConfigsList
 	} else {
-		state.TuningConfigs = types.ListNull(types.StringType)
+		if !common.HasValue(state.TuningConfigs) {
```
Comment (ciaranRoche) [inline provider/machinepool/hcp/machine_pool_resource.go:1193]: Does this need to be wrapped in the else

Comment (gdbranco) [reply provider/machinepool/hcp/machine_pool_resource.go]: Gotcha, I think we need similar refactors throughout then

Comment (ciaranRoche) [reply provider/machinepool/hcp/machine_pool_resource.go]: not a blocker, just NP, I wouldnt roll with making more refactors, functionally its good as is

**Pull request merged**